custom_connectors/oauth2/trackvia.rb

```diff
@@ -91,18 +91,22 @@
   methods: {
     get_all_output_fields: lambda do |input|
       url = input[:view_id].present? ? "views/#{input[:view_id]}" : "users"
+      relationshipForeignKeys = []
       Array.wrap(get(url)&.[]('structure')).reject { |field| field['name'] == "ID" }.map do |field|
         case field['type']
-        when 'number', 'currency', 'percentage', 'autoIncrement', 'relationship'
+        when 'relationship'
+          relationshipForeignKeys.push({ name: "f_#{field['fieldMetaId']}(id)", label: "#{field['name']}(id)", field_name: "#{field['name']}(id)", type: 'number' })
+          { name: "f_#{field['fieldMetaId']}", label: field['name'], field_name: field['name'], type: 'string' }
+        when 'number', 'currency', 'percentage', 'autoIncrement',
```

Comment:
@johnhuffmanattrackvia One last thing to make sure it doesn't break things. Previously "relationship" type fields were shown as numbers. Was there any way that users could have used this field? This caters for people who might have their recipes break if we push these changes.

Reply:
The previous design was technically incorrect.  In the api that powers this 2 values come through for a relationship, one which is the trackvia "Record ID" which, while having ID in the sa
[sampleResponse.txt](https://github.com/workato/custom_connector_docs/files/7014878/sampleResponse.txt)
me, is really a human friendly label and is most often a string.  The other is the fK and that is a number.  Let me attach a sample response to clarify this.


In that response, in the data property, note this value

`            "Link to Relationship Parent": "asdf123123123"
`
That property is the human friendly label referred to as "Record ID"

while this value 

`            "Link to Relationship Parent(id)": 4,
`

refers to the actual fK into the parent table.  the (id) is set on all fK values

Does that make it a little more clear what I'm trying to do?

Reply:
@johnhuffmanattrackvia So just to be sure, this change will not impact any existing trackvia users right? Before this, fields with type "relationship" could not be used in the connector right?

Reply:
@bennettgo they can be and they are used frequently.  The most recent change shouldn't change the relationship behavior that currently exists, only adds a new int field.  Do you see something that would break the current implementation?